gnovm/pkg/doc/json_doc.go

```diff
@@ -0,0 +1,246 @@
+package doc
+
+import (
+	"fmt"
+	"go/ast"
+	"go/doc"
+	"go/format"
+	"go/token"
+	"strings"
+
+	"github.com/gnolang/gno/gnovm"
+	"github.com/gnolang/gno/tm2/pkg/amino"
+)
+
+// JSONDocumentation holds package documentation suitable for transmitting
+// as JSON with printable string fields
+type JSONDocumentation struct {
+	PackagePath string `json:"package_path"`
+	PackageLine string `json:"package_line"` // package io // import "io"
+	PackageDoc  string `json:"package_doc"`  // markdown of top-level package documentation
+	// https://pkg.go.dev/go/doc#Package.Markdown to render markdown
+
+	// These match each of the sections in a pkg.go.dev package documentationj
+	Values []*JSONValueDecl `json:"values"` // constants and variables declared
+	Funcs  []*JSONFunc      `json:"funcs"`  // Funcs and methods
+	Types  []*JSONType      `json:"types"`
+}
+
+type JSONValueDecl struct {
+	Signature string       `json:"signature"`
+	Const     bool         `json:"const"`
+	Values    []*JSONValue `json:"values"`
+	Doc       string       `json:"doc"` // markdown
+}
+
+type JSONValue struct {
+	Name string `json:"name"`
+	Doc  string `json:"doc"`
+	Type string `json:"type"` // often empty
+}
+
+type JSONField struct {
+	Name string
+	Type string
+}
+
+type JSONFunc struct {
+	Type      string       `json:"type"` // if this is a method
+	Name      string       `json:"name"`
+	Signature string       `json:"signature"`
+	Doc       string       `json:"doc"` // markdown
+	Params    []*JSONField `json:"params"`
+	Results   []*JSONField `json:"results"`
+}
+
+type JSONType struct {
+	Name      string `json:"name"`
+	Signature string `json:"signature"`
+	Doc       string `json:"doc"` // markdown
+}
+
+// NewDocumentableFromMemPkg gets the pkgData from memPkg and returns a Documentable
+func NewDocumentableFromMemPkg(memPkg *gnovm.MemPackage, unexported bool, symbol, accessible string) (*Documentable, error) {
```

Comment:
Maybe the last fields would benefit from being in a type of their own, but let's leave as-is for now.

Reply:
OK.